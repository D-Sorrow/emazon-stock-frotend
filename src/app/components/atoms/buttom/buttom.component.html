<div class="btn-container">
    <button class="btn-container__btn" [ngClass]="buttonClass">
      <span class="btn__text">{{textButton}}</span>
    </button>
  </div>