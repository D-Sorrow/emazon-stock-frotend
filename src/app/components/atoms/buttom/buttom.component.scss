@import '../../../shared/tokens/colors';
@import '../../../shared/tokens/typografhy';

.btn-container {
    display: flex;
    justify-content: center;
    align-items: center;
  }

.btn-container__btn{
    cursor: pointer;
    border: none;
    background-color: $color_purple;
    padding: 10px;
    color: $color_white;
    border-radius: 4px;
    font-size: $main_text_body_l_regular_size;
    line-height: $main_text_body_l_line_height;
    font-weight: $main_text_font_weight;

    &:hover {
      background-color: $color_purple_hover;
    }
}