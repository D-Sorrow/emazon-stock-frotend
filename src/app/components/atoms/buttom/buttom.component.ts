import { Component, OnInit, Input } from '@angular/core';

@Component({
  selector: 'app-buttom',
  templateUrl: './buttom.component.html',
  styleUrls: ['./buttom.component.scss']
})
export class ButtomComponent implements OnInit {

  @Input() disabled: boolean = false;
  @Input() textButton: string ='';
  @Input() buttonClass: string = '';

  constructor() { }

  ngOnInit(): void {
    
  }

}
