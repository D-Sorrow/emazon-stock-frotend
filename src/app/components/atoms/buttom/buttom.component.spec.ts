import { ComponentFixture, TestBed } from '@angular/core/testing';

import { ButtomComponent } from './buttom.component';

describe('ButtomComponent', () => {
  let component: ButtomComponent;
  let fixture: ComponentFixture<ButtomComponent>;
  let compiled: HTMLElement;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      declarations: [ ButtomComponent ]
    })
    .compileComponents();

    fixture = TestBed.createComponent(ButtomComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
    compiled = fixture.nativeElement;
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('Matches with the component', () => {
    expect(compiled).toMatchSnapshot();
  });
});
