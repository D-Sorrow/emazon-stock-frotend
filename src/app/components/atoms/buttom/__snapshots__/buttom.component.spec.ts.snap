// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtomComponent Matches with the component 1`] = `
<div
  id="root1"
>
  <div
    class="btn-container"
  >
    <button
      class="btn-container__btn"
    >
      <span
        class="btn__text"
      >
        
      </span>
    </button>
  </div>
</div>
`;
